benchmarks/pnp_diode/diode.h

```diff
@@ -8,71 +8,109 @@
  * along with dimensional analysis
  */
 
-/**
- * dimensional analysis
- */
-const double elementary_charge = 1.60217662e-19; // C
-const double boltzmann = 1.38064852e-23; // J / K
+// Constants
+const double HALF_PI = 1.57079632679;
+const double ELEMENTARY_CHARGE = 1.60217662e-19; // C
+const double BOLTZMANN = 1.38064852e-23; // J / K
+const double VACUUM_PERMITTIVITY = 8.854187817e-12; // C / V*m
+
+// device
 const double temperature = 3e+2; // K
-const double thermodynamic_beta = elementary_charge / (temperature * boltzmann); // V
-const double vacuum_permittivity = 8.854187817e-12; // C / V
+const double thermodynamic_beta = ELEMENTARY_CHARGE / (temperature * BOLTZMANN); // C/J = 1/V
+const double relative_permittivity = 1.17e+1; // for silicon
 
-const double reference_length = 1e-5; // m
-const double reference_density = 1.5e+22; // mM = 1 / m^3
-const double reference_diffusivity = 2.87e+1; // cm^2 / s
-const double reference_relative_permittivity = 1.17e+1; // dimensionless
+// maj * min = intrinsic carrier density = 1e+16 / m^3
+// maj - min = - doping level
+// const double majority_carrier = 1.5e+22; // 1 / m^3
+// const double minority_carrier = 6.66667e+9; // 1 / m^3
 
-const double permittivity_factor = reference_relative_permittivity * vacuum_permittivity /
-  (elementary_charge * thermodynamic_beta * reference_density * reference_length * reference_length);
+// const double majority_carrier = 5.0e+20;
+// const double minority_carrier = 2.0e+10;
 
-double scale_density (double density) { return density / reference_density; };
-double scale_potential (double phi) { return (phi) * thermodynamic_beta; };
-double scale_diffusivity (double diff) { return diff / reference_diffusivity; };
-double scale_rel_permittivity (double rel_perm) { return rel_perm / reference_relative_permittivity; };
+const double majority_carrier = 1.0e+19;
+const double minority_carrier = 9.999990000020e+11;
```

Comment:
@arthbous 
Using values from lines 24-25 is a much harder linear system
Using values from lines 27-28 is a harder linear system
Using 30-31 is easier.